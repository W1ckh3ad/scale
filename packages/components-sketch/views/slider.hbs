<div data-sketch-artboard="Slider Track Small" class="artboard">
    <h1 class="title--artboard">Slider Track Small</h1>

    <h2 class="title--artboard-section">Label On - Counter On</h2>
    <div class="agent-states--vertical-grid">
        <div class="agent-states--item">
            <scale-slider label="Slider Label" value="40" track-height="3"></scale-slider>
        </div>
        <div class="agent-states--item">
            magenta
        </div>
        <div class="agent-states--item">
            <scale-slider label="Slider Label" value="40" track-height="3" custom-color="#009DE0"></scale-slider>
        </div>
        <div class="agent-states--item">
          custom color
        </div>
        <div class="agent-states--item">
          <scale-slider label="Slider Label" value="40" track-height="3" disabled></scale-slider>
        </div>
        <div class="agent-states--item">
          disabled
        </div>
    </div>

    <h2 class="title--artboard-section">Label On - Counter Off</h2>
    <div class="agent-states--vertical-grid">
        <div class="agent-states--item">
            <scale-slider label="Slider Label" value="40" show-value="false" track-height="3"></scale-slider>
        </div>
        <div class="agent-states--item">
            magenta
        </div>
        <div class="agent-states--item">
            <scale-slider label="Slider Label" value="40" show-value="false" track-height="3" custom-color="#009DE0"></scale-slider>
        </div>
        <div class="agent-states--item">
          custom color
        </div>
        <div class="agent-states--item">
          <scale-slider label="Slider Label" value="40" show-value="false" track-height="3" disabled></scale-slider>
        </div>
        <div class="agent-states--item">
          disabled
        </div>
    </div>

    <h2 class="title--artboard-section">Label Off - Counter On</h2>
    <div class="agent-states--vertical-grid">
        <div class="agent-states--item">
            <scale-slider value="40" track-height="3"></scale-slider>
        </div>
        <div class="agent-states--item">
            magenta
        </div>
        <div class="agent-states--item">
            <scale-slider value="40" track-height="3" custom-color="#009DE0"></scale-slider>
        </div>
        <div class="agent-states--item">
          custom color
        </div>
        <div class="agent-states--item">
          <scale-slider value="40" track-height="3" disabled></scale-slider>
        </div>
        <div class="agent-states--item">
          disabled
        </div>
    </div>

    <h2 class="title--artboard-section">Label Off - Counter Off</h2>
    <div class="agent-states--vertical-grid">
        <div class="agent-states--item">
            <scale-slider value="40" show-value="false" track-height="3"></scale-slider>
        </div>
        <div class="agent-states--item">
            magenta
        </div>
        <div class="agent-states--item">
            <scale-slider value="40" show-value="false" track-height="3" custom-color="#009DE0"></scale-slider>
        </div>
        <div class="agent-states--item">
          custom color
        </div>
        <div class="agent-states--item">
          <scale-slider value="40" show-value="false" track-height="3" disabled></scale-slider>
        </div>
        <div class="agent-states--item">
          disabled
        </div>
    </div>
</div>

<div data-sketch-artboard="Slider Track Large" class="artboard">
    <h1 class="title--artboard">Slider Track Large</h1>

    <h2 class="title--artboard-section">Label On - Counter On</h2>
    <div class="agent-states--vertical-grid">
        <div class="agent-states--item">
            <scale-slider label="Slider Label" value="40" track-height="5"></scale-slider>
        </div>
        <div class="agent-states--item">
            magenta
        </div>
        <div class="agent-states--item">
            <scale-slider label="Slider Label" value="40" track-height="5" custom-color="#009DE0"></scale-slider>
        </div>
        <div class="agent-states--item">
          custom color
        </div>
        <div class="agent-states--item">
          <scale-slider label="Slider Label" value="40" track-height="5" disabled></scale-slider>
        </div>
        <div class="agent-states--item">
          disabled
        </div>
    </div>

    <h2 class="title--artboard-section">Label On - Counter Off</h2>
    <div class="agent-states--vertical-grid">
        <div class="agent-states--item">
            <scale-slider label="Slider Label" value="40" show-value="false" track-height="5"></scale-slider>
        </div>
        <div class="agent-states--item">
            magenta
        </div>
        <div class="agent-states--item">
            <scale-slider label="Slider Label" value="40" show-value="false" track-height="5" custom-color="#009DE0"></scale-slider>
        </div>
        <div class="agent-states--item">
          custom color
        </div>
        <div class="agent-states--item">
          <scale-slider label="Slider Label" value="40" show-value="false" track-height="5" disabled></scale-slider>
        </div>
        <div class="agent-states--item">
          disabled
        </div>
    </div>

    <h2 class="title--artboard-section">Label Off - Counter On</h2>
    <div class="agent-states--vertical-grid">
        <div class="agent-states--item">
            <scale-slider value="40" track-height="5"></scale-slider>
        </div>
        <div class="agent-states--item">
            magenta
        </div>
        <div class="agent-states--item">
            <scale-slider value="40" track-height="5" custom-color="#009DE0"></scale-slider>
        </div>
        <div class="agent-states--item">
          custom color
        </div>
        <div class="agent-states--item">
          <scale-slider value="40" track-height="5" disabled></scale-slider>
        </div>
        <div class="agent-states--item">
          disabled
        </div>
    </div>

    <h2 class="title--artboard-section">Label Off - Counter Off</h2>
    <div class="agent-states--vertical-grid">
        <div class="agent-states--item">
            <scale-slider value="40" show-value="false" track-height="5"></scale-slider>
        </div>
        <div class="agent-states--item">
            magenta
        </div>
        <div class="agent-states--item">
            <scale-slider value="40" show-value="false" track-height="5" custom-color="#009DE0"></scale-slider>
        </div>
        <div class="agent-states--item">
          custom color
        </div>
        <div class="agent-states--item">
          <scale-slider value="40" show-value="false" track-height="5" disabled></scale-slider>
        </div>
        <div class="agent-states--item">
          disabled
        </div>
    </div>
</div>

<script src="/symbol_names.js"></script>
<script>
    Array.from(document.querySelectorAll("scale-slider")).forEach(btn => {
      /*
        btn.dataset.sketchSymbol = `Switch / ${
                order(1, btn.hasAttribute('active') ? 'Selected On' : 'Selected Off')
            } / ${
                order(4, getStateName(btn))
            }`;
      */
    });
</script>