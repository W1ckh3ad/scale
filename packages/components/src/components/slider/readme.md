# scale-slider



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                                                           | Type      | Default     |
| ------------- | -------------- | --------------------------------------------------------------------- | --------- | ----------- |
| `customClass` | `custom-class` | (optional) Slider class                                               | `string`  | `''`        |
| `customColor` | `custom-color` | (optional) slider custom color                                        | `string`  | `''`        |
| `disabled`    | `disabled`     | (optional) disabled                                                   | `boolean` | `false`     |
| `label`       | `label`        | (optional) slider label                                               | `string`  | `undefined` |
| `max`         | `max`          | (optional) the maximal value of the slider                            | `number`  | `100`       |
| `min`         | `min`          | t(optional) he minimal value of the slider                            | `number`  | `0`         |
| `showValue`   | `show-value`   | (optional) slider display value                                       | `boolean` | `true`      |
| `step`        | `step`         | (optional) the step size to increase or decrease when dragging slider | `number`  | `1`         |
| `styles`      | `styles`       | (optional) Injected jss styles                                        | `any`     | `undefined` |
| `trackHeight` | `track-height` | (optional) the height of slider track                                 | `number`  | `undefined` |
| `value`       | `value`        | (optional) the display value of the slider                            | `number`  | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
