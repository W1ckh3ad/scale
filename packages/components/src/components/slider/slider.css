.slider {
  width: 100%;
  display: block;
  align-items: center;
}

.slider .slider__track-wrapper {
  display: flex;
  align-items: center;
}

.slider .slider__track {
  width: 303px;
  border: 1px solid var(--scl-color-grey-60);
  height: 6px;
  margin: 16px 0;
  display: flex;
  position: relative;
  box-sizing: border-box;
  align-items: center;
  border-radius: 100px;
}

.slider .slider__bar {
  height: 6px;
  z-index: -1;
  position: absolute;
  border-radius: 100px;
  background-color: var(--scl-color-primary);
}

.slider .slider__thumb-wrapper {
  width: 32px;
  height: 32px;
  display: flex;
  z-index: 100;
  position: absolute;
  text-align: center;
  align-items: center;
  margin-left: -16px;
  justify-content: center;
  background-color: transparent;
}

.slider .slider__thumb {
  width: 16px;
  border: 1px solid;
  height: 16px;
  outline: none;
  box-sizing: border-box;
  border-color: var(--scl-color-grey-60);
  border-radius: 50%;
  background-color: #fff;
  box-shadow: var(--scl-shadow-level-0);
}

.slider .slider__display-value {
  color: var(--scl-color-grey-60);
  margin-left: 24px;
  font-weight: var(--scl-font-weight-bold);
  font-size: var(--scl-font-size-12);
}

.slider .slider__thumb:hover {
  border-color: var(--scl-color-primary-hover);
}

.slider .slider__thumb:active {
  border-color: var(--scl-color-primary-active);
}

.slider .slider__thumb:focus {
  box-shadow:  0 0 0 var(--scl-spacing-2) var(--scl-color-focus);
}

.slider .slider__thumb-wrapper:hover {
  cursor: grab;
}

.slider .slider__thumb-wrapper:active {
  cursor: grabbing;
}

.slider--track-small .slider__track {
  border: none;
  height: 1px;
  border-top: 1px solid transparent;
  background-color: var(--scl-color-grey-50);
}

.slider--track-small .slider__bar {
  border: 1px solid transparent;
  height: 3px;
  z-index: 1;
  box-sizing: border-box;
}

.slider--thumb-large .slider__thumb {
  width: 24px;
  height: 24px;
}

.slider--disabled .slider__track-wrapper {
  cursor: not-allowed;
}

.slider--disabled .slider__bar {
  background-color: var(--scl-color-grey-50);
}

.slider--disabled .slider__thumb {
  display: none;
}

.slider--disabled .slider__label {
  color: var(--scl-color-grey-50);
}

.slider--disabled .slider__thumb-wrapper:hover {
  cursor: not-allowed;
}
