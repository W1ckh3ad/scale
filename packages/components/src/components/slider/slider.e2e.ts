import { newE2EPage } from '@stencil/core/testing';

describe('scale-slider', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-slider/>');
    const element = await page.find('scale-slider');
    expect(element).toHaveClass('hydrated');
  });
});
