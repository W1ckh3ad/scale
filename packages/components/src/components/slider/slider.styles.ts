import { JssStyle } from 'jss';

export const styles: JssStyle = {
  slider: {
    display: 'block',
    alignItems: 'center',
    width: '100%',
    '& .slider--track-wrapper': {
      display: 'flex',
      alignItems: 'center',
    },
    '& .slider--track': {
      boxSizing: 'border-box',
      width: '303px', // slider track size should have a fix number not percentage otherwise when dragging, the track flickers
      height: 6,
      margin: '16px 0',
      border: '1px solid #6c6c6c',
      borderRadius: '100px',
      position: 'relative',
      display: 'flex',
      alignItems: 'center',
    },
    '& .slider--bar': {
      height: 6,
      backgroundColor: ({ color }) => color.primary,
      borderRadius: '100px',
      position: 'absolute',
      zIndex: -1,
    },
    '& .slider--thumb-wrapper': {
      height: 32,
      width: 32,
      position: 'absolute',
      zIndex: 100,
      marginLeft: -16,
      backgroundColor: 'transparent',
      textAlign: 'center',
      userSelect: 'none',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      '&:hover': {
        cursor: 'grab',
      },
      '&:active': {
        cursor: 'grabbing',
      },
    },
    '& .slider--thumb': {
      boxSizing: 'border-box',
      width: 16,
      height: 16,
      border: '1px solid',
      borderColor: '#6c6c6c',
      backgroundColor: '#fff',
      borderRadius: '50%',
      userSelect: 'none',
      boxShadow: ({ shadow }) => shadow.ground,
      '&:hover': {
        borderColor: ({ color }) => color.primary_hover,
      },
      '&:active': {
        borderColor: ({ color }) => color.primary_active,
      },
    },
    '& .slider--display-value': {
      marginLeft: 24,
      fontFamily: 'TeleNeoWeb-Bold',
      fontSize: 12,
      color: '#6c6c6c',
    },
  },
  'slider--track-small': {
    '& .slider--track': {
      height: 1,
      border: 'none',
      backgroundColor: '#7c7c7c',
    },
    '& .slider--bar': {
      height: 3,
      zIndex: 1,
    },
  },
  'slider--thumb-large': {
    '& .slider--thumb': {
      width: 24,
      height: 24,
    },
  },
  'slider--disabled': {
    '& .slider--track-wrapper': {
      cursor: 'not-allowed',
    },
    '& .slider--bar': {
      backgroundColor: '#7c7c7c',
    },
    '& .slider--thumb': {
      display: 'none',
    },
    '& .slider--thumb-wrapper': {
      '&:hover': {
        cursor: 'not-allowed',
      },
    },
  },
};
