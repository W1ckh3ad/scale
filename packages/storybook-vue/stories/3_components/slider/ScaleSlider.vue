<template>
  <scale-slider
    :custom-color="customColor"
    :disabled="disabled"
    :label="label"
    :max="max"
    :min="min"
    :show-value="showValue"
    :track-small="trackSmall"
    :thumb-large="thumbLarge"
    :step="step"
    :value="value"
  >
  </scale-slider>
</template>

<script>
export default {
  props: {
    customColor: String,
    disabled: { type: Boolean, default: false },
    label: String,
    max: { type: Number, default: 100 },
    min: { type: Number, default: 0 },
    showValue: { type: Boolean, default: true },
    trackSmall: { type: Boolean, default: false },
    thumbLarge: { type: Boolean, default: false },
    step: { type: Number, default: 1 },
    value: Number
  }
};
</script>
