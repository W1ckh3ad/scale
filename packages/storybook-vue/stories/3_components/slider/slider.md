# Slider

Sliders provide a visual indication of adjustable content. The user can increase or decrease a value by moving the handle along a horizontal track.

## General

Sliders allow users to view and select a value from a given range along a bar. They are ideal for adjusting settings such as volume and brightness or for applying filters. Changes made with sliders are immediate, allowing users to make adjustments while determining their selection. Sliders should not be used for settings where user feedback comes with delay.

## Basic Types

### Value

![Image Name](/assets/3_components/slider/image-20200810093330831.png)

### Range

![Image Name](/assets/3_components/slider/image-20200810093342268.png)

## Elements

In its basic form, the slider is accompanied by a label and a number displaying the current value.

## Colors

Use the default color or assign a custom color. When customizing, use a color-coding scheme that is clear to the user and supports them in their use cases.

![Image Name](/assets/3_components/slider/image-20200810093401089.png)

## Sizes

### Large

![Image Name](/assets/3_components/slider/image-20200810093350850.png)

### Small

![Image Name](/assets/3_components/slider/image-20200810093330831.png)

## Best Practice

- The slider label should indicate what value the slider is changing.
- Range values are used to describe the range in numbers.
- Do not use for extremely large ranges i.e. 1-1000.
- Do not use for extremely small ranges i.e. 1-3.

## Related Components
