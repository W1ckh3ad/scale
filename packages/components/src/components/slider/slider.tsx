import {
  Component,
  h,
  State,
  Prop,
  Host,
  Event,
  EventEmitter,
} from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './slider.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';

let i = 0;

@Component({
  tag: 'scale-slider',
  shadow: true,
})
export class Slider implements Base {
  sliderTrack?: HTMLDivElement;

  /** (optional) Slider class */
  @Prop() customClass?: string = '';
  /** (optional) the display value of the slider */
  @Prop() value?: number;
  /** t(optional) he minimal value of the slider */
  @Prop() min?: number = 0;
  /** (optional) the maximal value of the slider */
  @Prop() max?: number = 100;
  /** (optional) the step size to increase or decrease when dragging slider */
  @Prop() step?: number = 1;
  /** (optional) slider label */
  @Prop() label?: string;
  /** (optional) slider display value */
  @Prop() showValue?: boolean = true;
  /** (optional) slider custom color */
  @Prop() customColor?: string = '';
  /** (optional) disabled  */
  @Prop() disabled?: boolean = false;

  /** (optional) smaller track */
  @Prop() trackSmall?: boolean = false;
  /** (optional) larger thumb */
  @Prop() thumbLarge?: boolean = false;

  /** (optional) Slider id */
  @Prop() sliderId?: string;

  /** (optional) Injected jss styles */
  @Prop() styles?: any;
  /** decorator Jss stylesheet */
  @CssInJs('Slider', styles) stylesheet: StyleSheet;

  @State() dragging: boolean;
  @State() startX: number;
  @State() currentX: number;
  @State() startPosition: number;
  @State() newPosition: number;

  @Event() scaleChange: EventEmitter<number>;

  constructor() {
    this.onDragging = this.onDragging.bind(this);
    this.onDragEnd = this.onDragEnd.bind(this);
  }

  componentWillLoad() {
    if (this.sliderId == null) {
      this.sliderId = 'slider-' + i++;
    }
  }
  componentWillUpdate() {}
  componentDidUnload() {
    this.removeGlobalListeners();
  }

  onButtonDown = (event: any) => {
    if (this.disabled) {
      return;
    }
    this.onDragStart(event);
    this.addGlobalListeners();
  };

  onDragStart = (event: any) => {
    this.dragging = true;
    this.startX = this.handleTouchEvent(event).clientX;
    this.startPosition = parseInt(this.currentPosition(), 10);
  };

  onDragging = (event: any) => {
    const { dragging, startX, startPosition } = this;

    if (dragging) {
      this.currentX = this.handleTouchEvent(event).clientX;

      let diff: number;

      diff = ((this.currentX - startX) / this.sliderTrack.offsetWidth) * 100;

      this.newPosition = startPosition + diff;
      this.setPosition(this.newPosition);
    }
  };

  onDragEnd = () => {
    const { dragging, newPosition } = this;
    if (dragging) {
      this.dragging = false;
    }
    this.setPosition(newPosition || this.startPosition);
    this.removeGlobalListeners();
  };

  addGlobalListeners() {
    window.addEventListener('mousemove', this.onDragging.bind(this));
    window.addEventListener('mouseup', this.onDragEnd.bind(this));
    window.addEventListener('touchmove', this.onDragging.bind(this));
    window.addEventListener('touchend', this.onDragEnd.bind(this));
  }

  removeGlobalListeners() {
    window.removeEventListener('mousemove', this.onDragging);
    window.removeEventListener('mouseup', this.onDragEnd);
    window.removeEventListener('touchmove', this.onDragging);
    window.removeEventListener('touchend', this.onDragEnd);
  }

  handleTouchEvent(event: any): MouseEvent | Touch {
    return event.type.indexOf('touch') === 0 ? event.touches[0] : event;
  }

  onKeyDown = event => {
    if (['ArrowRight', 'ArrowLeft'].includes(event.key)) {
      this.setPosition(
        this.value + (event.key === 'ArrowRight' ? this.step : -this.step)
      );
    }
    if (['ArrowUp', 'ArrowDown'].includes(event.key)) {
      this.setPosition(
        this.value +
          (event.key === 'ArrowUp' ? this.step * 10 : -this.step * 10)
      );
    }
  };

  setPosition = (newPosition: number) => {
    if (newPosition < 0) {
      newPosition = 0;
    } else if (newPosition > 100) {
      newPosition = 100;
    }

    const lengthPerStep = 100 / ((this.max - this.min) / this.step);
    const steps = Math.round(newPosition / lengthPerStep);
    this.value =
      steps * lengthPerStep * (this.max - this.min) * 0.01 + this.min;
    this.scaleChange.emit(Math.abs(this.value));
  };

  currentPosition(): string {
    return `${((this.value - this.min) / (this.max - this.min)) * 100}%`;
  }

  render() {
    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <div class={this.getCssClassMap()}>
          {!!this.label && (
            <label
              class="slider--label"
              id={`${this.sliderId}-label`}
              htmlFor={this.sliderId}
            >
              {this.label}
            </label>
          )}
          <div class="slider--track-wrapper">
            <div
              class="slider--track"
              ref={el => (this.sliderTrack = el as HTMLDivElement)}
            >
              <div
                class="slider--bar"
                style={{
                  width: `${this.value}%`,
                  backgroundColor: this.customColor,
                }}
              ></div>
              <div
                class="slider--thumb-wrapper"
                style={{ left: `${this.value}%` }}
                onMouseDown={this.onButtonDown}
                onTouchStart={this.onButtonDown}
              >
                <div
                  class="slider--thumb"
                  tabindex="0"
                  role="slider"
                  id={this.sliderId}
                  aria-valuemin={this.min}
                  aria-valuenow={this.value}
                  aria-valuemax={this.max}
                  aria-valuetext={`${this.value}%`}
                  aria-labelledby={`${this.sliderId}-label`}
                  aria-orientation="horizontal"
                  aria-disabled={this.disabled}
                  onKeyDown={this.onKeyDown}
                />
              </div>
            </div>
            {this.showValue && (
              <div class="slider--display-value">{this.value}%</div>
            )}
          </div>
        </div>
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(
      classes.slider,
      this.customClass && this.customClass,
      this.disabled && classes[`slider--disabled`],
      this.trackSmall && classes[`slider--track-small`],
      this.thumbLarge && classes[`slider--thumb-large`]
    );
  }
}
