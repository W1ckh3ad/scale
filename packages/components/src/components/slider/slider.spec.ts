import { newSpecPage } from '@stencil/core/testing';
import { Slider } from './slider';
import { styles } from './slider.styles';
import jss from 'jss';

describe('Slider', () => {
  let element;
  beforeEach(async () => {
    element = new Slider();
    element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Slider],
      html: `<scale-slider>Label</scale-slider>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');
  });
});
