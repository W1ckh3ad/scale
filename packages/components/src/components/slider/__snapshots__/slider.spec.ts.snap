// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should match snapshot 1`] = `
<scale-slider>
  <mock:shadow-root>
    <div class="slider" part="slider">
      <div class="slider__track-wrapper" part="track-wrapper">
        <div class="slider__track" part="track">
          <div class="slider__bar" part="bar" style="width: undefined%;"></div>
          <div class="slider__thumb-wrapper" part="thumb-wrapper" style="left: undefined%;">
            <div aria-labelledby="slider-0-label" aria-orientation="horizontal" aria-valuemax="100" aria-valuemin="0" aria-valuetext="undefined%" class="slider__thumb" id="slider-0" part="thumb" role="slider" tabindex="0"></div>
          </div>
        </div>
        <div class="slider__display-value" part="display-value">
          %
        </div>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-slider>
`;
